tests/test_snql_sdk_api.py

```diff
@@ -0,0 +1,163 @@
+            minute=0, second=0, microsecond=0
+        ) - timedelta(minutes=180)
+        events_storage = get_entity(EntityKey.EVENTS).get_writable_storage()
+        write_unprocessed_events(events_storage, [self.event])
+        self.next_time = datetime.utcnow().replace(
+            minute=0, second=0, microsecond=0
+        ) + timedelta(minutes=180)
+        write_unprocessed_events(
+            get_writable_storage(StorageKey.TRANSACTIONS), [get_raw_transaction()],
+        )
+
+    def test_simple_query(self) -> None:
+        query = (
+            Query("discover", Entity("discover_events"))
+            .set_select([Function("count", [], "count")])
+            .set_groupby([Column("project_id"), Column("tags[custom_tag]")])
+            .set_where(
+                [
+                    Condition(Column("type"), Op.NEQ, "transaction"),
+                    Condition(Column("project_id"), Op.EQ, self.project_id),
+                    Condition(Column("timestamp"), Op.GTE, self.base_time),
+                    Condition(Column("timestamp"), Op.LT, self.next_time),
+                ]
+            )
+            .set_orderby([OrderBy(Function("count", [], "count"), Direction.ASC)])
+            .set_limit(1000)
+            .set_consistent(True)
+            .set_debug(True)
+        )
+
+        response = self.app.post("/discover/snql", data=query.snuba())
+        data = json.loads(response.data)
+
+        assert response.status_code == 200, data
+        assert data["stats"]["consistent"]
+        assert data["data"] == [
+            {
+                "count": 1,
+                "tags[custom_tag]": "custom_value",
+                "project_id": self.project_id,
+            }
+        ]
+
+    def test_sessions_query(self) -> None:
+        query = (
+            Query("sessions", Entity("sessions"))
+            .set_select([Column("project_id"), Column("release")])
+            .set_groupby([Column("project_id"), Column("release")])
+            .set_where(
+                [
+                    Condition(Column("project_id"), Op.IN, [self.project_id]),
+                    Condition(Column("org_id"), Op.EQ, self.org_id),
+                    Condition(
+                        Column("started"),
+                        Op.GTE,
+                        datetime(2021, 1, 1, 17, 5, 59, 554860),
+                    ),
+                    Condition(
+                        Column("started"), Op.LT, datetime(2022, 1, 1, 17, 6, 0, 554981)
+                    ),
+                ]
+            )
+            .set_orderby([OrderBy(Column("sessions"), Direction.DESC)])
+            .set_limit(100)
+        )
+
+        response = self.app.post("/sessions/snql", data=query.snuba())
+        data = json.loads(response.data)
+
+        assert response.status_code == 200
+        assert data["data"] == []
+
+    def test_join_query(self) -> None:
+        sp = Entity("spans", "s")
+        tr = Entity("transactions", "tr")
```

Comment:
We are getting rid of the span entity soon. I would use events and groupedmessage.